runtimevar/constantvar/constantvar.go

```diff
@@ -24,13 +37,51 @@ package constantvar // import "gocloud.dev/runtimevar/constantvar"
 import (
 	"context"
 	"errors"
+	"fmt"
+	"net/url"
 	"time"
 
 	"gocloud.dev/gcerrors"
 	"gocloud.dev/runtimevar"
 	"gocloud.dev/runtimevar/driver"
 )
 
+func init() {
+	runtimevar.DefaultURLMux().RegisterVariable(Scheme, &urlOpener{})
+}
+
+// Scheme is the URL scheme constantvar registers its URLOpener under on blob.DefaultMux.
+const Scheme = "constant"
+
+type urlOpener struct{}
+
+func (*urlOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	q := u.Query()
+	decoder, err := runtimevar.DecoderByName(q.Get("decoder"))
+	if err != nil {
+		return nil, fmt.Errorf("open variable %q: invalid \"decoder\": %v", u, err)
+	}
+	var value string
+	var errVal error
+	for param, values := range q {
+		val := values[0]
+		switch param {
+		case "decoder":
+			// processed elsewhere
+		case "val":
+			value = val
+		case "err":
```

Comment:
If the URL contains both "val" and "err", it returns the error, I think we should document this.

Reply:
Done.